benchmarks/src/main/scala/benchmark/MergeBenchmark.scala

```diff
@@ -0,0 +1,148 @@
+  def userDefinedDbName: Option[String]
+  def dbName: String = userDefinedDbName.getOrElse(s"merge_sf${scaleInGB}")
+  def dbLocation: String = dbLocation(dbName)
+}
+
+case class MergeBenchmarkConf(
+     scaleInGB: Int = 0,
+     userDefinedDbName: Option[String] = None,
+     iterations: Int = 3,
+     benchmarkPath: Option[String] = None) extends MergeConf {
+}
+
+object MergeBenchmarkConf {
+  import scopt.OParser
+  private val builder = OParser.builder[MergeBenchmarkConf]
+  private val argParser = {
+    import builder._
+    OParser.sequence(
+      programName("Merge Benchmark"),
+      opt[String]("scale-in-gb")
+        .required()
+        .valueName("<scale of benchmark in GBs>")
+        .action((x, c) => c.copy(scaleInGB = x.toInt))
+        .text("Scale factor of the TPCDS benchmark"),
+      opt[String]("benchmark-path")
+        .required()
+        .valueName("<cloud storage path>")
+        .action((x, c) => c.copy(benchmarkPath = Some(x)))
+        .text("Cloud path to be used for creating table and generating reports"),
+      opt[String]("iterations")
+        .optional()
+        .valueName("<number of iterations>")
+        .action((x, c) => c.copy(iterations = x.toInt))
+        .text("Number of times to run the queries"),
+    )
+  }
+
+  def parse(args: Array[String]): Option[MergeBenchmarkConf] = {
+    OParser.parse(argParser, args, MergeBenchmarkConf())
+  }
+}
+
+class MergeBenchmark(conf: MergeBenchmarkConf) extends Benchmark(conf) {
+  val extraConfs: Map[String, String] = Map(
+    "spark.sql.broadcastTimeout" -> "7200",
+    "spark.sql.crossJoin.enabled" -> "true"
+  )
+
+  /**
+   * Runs every merge test case multiple times and records the duration.
+   */
+  override def runInternal(): Unit = {
+    for ((k, v) <- extraConfs) spark.conf.set(k, v)
+    spark.sparkContext.setLogLevel("WARN")
+    log("All configs:\n\t" + spark.conf.getAll.toSeq.sortBy(_._1).mkString("\n\t"))
+    spark.sql(s"USE ${conf.dbName}")
+
+    val targetRowCount = spark.read.table(s"`${conf.dbName}`.`target_${conf.tableName}`").count
+
+    for (iteration <- 1 to conf.iterations) {
+      MergeTestCases.testCases.foreach { runMerge(_, targetRowCount, iteration = Some(iteration)) }
+    }
+    val results = getQueryResults().filter(_.name.startsWith("q"))
+    if (results.forall(x => x.errorMsg.isEmpty && x.durationMs.nonEmpty) ) {
+      val medianDurationSecPerQuery = results.groupBy(_.name).map { case (q, results) =>
+        assert(results.size == conf.iterations)
+        val medianMs = results.map(_.durationMs.get).sorted
+            .drop(math.floor(conf.iterations / 2.0).toInt).head
+        (q, medianMs / 1000.0)
+      }
+      val sumOfMedians = medianDurationSecPerQuery.map(_._2).sum
+      reportExtraMetric("merge-result-seconds", sumOfMedians)
+    }
+  }
+
+  /**
+   * Merge test runner: clone a fresh target table, run the merge test case, checks invariants then
+   * drops the cloned table.
```

Comment:
nit: check invariants, then drop the cloned table. It would be better as a list, e.g:

- Clone a fresh target table.
- Run the merge test case.
- Check invariants.
- Drop the cloned table.

Reply:
Done